File: leetcode_solutions/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        prefix = strs[0]
        for word in strs:
            i = 0
            if word != prefix:
                while i < len(word) and i < len(prefix):
                    if word[i] != prefix[i]:
                        prefix = prefix[0:i]
                        break
                    i+=1
        return prefix

File: leetcode_solutions/0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        int n = strs.length - 1;
        StringBuilder ans = new StringBuilder();

        String first = strs[0];
        String last = strs[n];

        for (int i = 0; i < Math.min(first.length(), last.length()); i++){
            if (first.charAt(i) != last.charAt(i)){
                return ans.toString();
            }
            ans.append(first.charAt(i));
        }

        return ans.toString();
    }
}
